internal/daemon/daemon.go

```diff
@@ -746,18 +884,48 @@ func (d *Daemon) ClusterCert() *shared.CertInfo {
 	return shared.NewCertInfo(d.clusterCert.KeyPair(), d.clusterCert.CA(), d.clusterCert.CRL())
 }
 
-// ReloadClusterCert reloads the cluster keypair from the state directory.
-func (d *Daemon) ReloadClusterCert() error {
+// ReloadCert reloads a specific certificate from the filesytem.
+func (d *Daemon) ReloadCert(name string) error {
 	d.clusterMu.Lock()
 	defer d.clusterMu.Unlock()
 
-	clusterCert, err := util.LoadClusterCert(d.os.StateDir)
+	var dir string
+	if name == internalTypes.ClusterCertificate {
+		dir = d.os.StateDir
+	} else {
+		dir = d.os.CertificatesDir
+	}
+
+	cert, err := shared.KeyPairAndCA(dir, name, shared.CertServer, true)
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to load TLS certificate %q: %w", name, err)
 	}
 
-	d.clusterCert = clusterCert
-	d.endpoints.UpdateTLS(clusterCert)
+	// In case the cluster certificate gets reloaded also populate its value.
+	if name == internalTypes.ClusterCertificate {
+		d.clusterCert = cert
+
+		// The core API endpoints are labeled with core.
+		// When the cluster certificate gets updated reload those.
+		d.endpoints.UpdateTLSByName("core", cert)
```

Comment:
I thought you were going to use `cluster` as the name for the core servers so that you could reuse `internalTypes.ClusterCertificate`?

Reply:
Jup but I decided against as we tend to label everything with core/core API so to make it more clear I didn't use `cluster` at the end.

Now by having a constant for both `core` and `unix` (as you suggested) it makes even more sense :)

Reply:
Fixed.